src/unix/core.c

```diff
@@ -1102,3 +1102,53 @@ int uv_os_homedir(char* buffer, size_t* size) {
 
   return 0;
 }
+
+
+int uv_os_tmpdir(char* buffer, size_t* size) {
+  char* buf;
+  size_t len;
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return -EINVAL;
+
+  /* Check the TMPDIR, TMP, TEMP, and TEMPDIR environment variables in order */
+  buf = getenv("TMPDIR");
+
+  if (buf != NULL)
+    goto return_buffer;
+
+  buf = getenv("TMP");
+
+  if (buf != NULL)
+    goto return_buffer;
+
+  buf = getenv("TEMP");
+
+  if (buf != NULL)
+    goto return_buffer;
+
+  buf = getenv("TEMPDIR");
+
+  if (buf != NULL)
+    goto return_buffer;
+
+  /* No temp environment variables defined */
+  #if defined(__ANDROID__)
+    buf = "/data/local/tmp";
+  #else
+    buf = "/tmp";
+  #endif
+
+return_buffer:
+  len = strlen(buf);
+
+  if (len >= *size) {
+    *size = len;
+    return -ENOBUFS;
+  }
+
+  memcpy(buffer, buf, len + 1);
```

Comment:
Some ancient unix APIs don't null terminate because the copy mem from internal fixed length data structures - they are pretty low-level and horrid to use, uv should do better.